streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -418,7 +417,7 @@ void closeSuspended(boolean clean, RuntimeException firstException) {
                     try {
                         producer.abortTransaction();
                         transactionInFlight = false;
-                    } catch (final ProducerFencedException e) {
+                    } catch (final KafkaException e) {
```

Comment:
As discussed with @mjsax offline, as the caller of the producer API we should not expect any exceptions except javadoc'ed `ProducerFencedException`; in addition, three exceptions today should be treated as fatal and the caller (Streams) should avoid call other functions than `close()` when encountering it:

```
 try {
     producer.beginTransaction();
     for (int i = 0; i < 100; i++)
         producer.send(new ProducerRecord<>("my-topic", Integer.toString(i), Integer.toString(i)));
     producer.commitTransaction();
 } catch (ProducerFencedException | OutOfOrderSequenceException | AuthorizationException e) {
     // We can't recover from these exceptions, so our only option is to close the producer and exit.
     producer.close();
 } catch (KafkaException e) {
     // For all other exceptions, just abort the transaction and try again.
     producer.abortTransaction();
 }
 producer.close();
```